File: trades/trades.js
const axios = require('axios');
const crypto = require('crypto');
require('dotenv').config();

async function fetchTrades(marketCode, limit, startTime, endTime) {
  const apiKey = process.env.API_KEY;
  const secretKey = process.env.API_SECRET;
  const ts = new Date().toISOString().split('.')[0] + 'Z';
  const nonce = Date.now().toString();
  const method = "/v3/trades";
  const apiUrl = "api.ox.fun";

  const queryString = `marketCode=${marketCode}&limit=${limit}&startTime=${startTime}&endTime=${endTime}`;
  const msgString = `${ts}\n${nonce}\nGET\n${apiUrl}\n${method}\n${queryString}`;

  const sign = crypto.createHmac('sha256', secretKey)
    .update(msgString)
    .digest('base64');

  const headers = {
    'Content-Type': 'application/json',
    'AccessKey': apiKey,
    'Timestamp': ts,
    'Signature': sign,
    'Nonce': nonce
  };

  const url = `https://${apiUrl}${method}?${queryString}`;

  try {
    const response = await axios.get(url, { headers });
    if (response.data.success) {
      console.log('Trades Data:', response.data.data);
    } else {
      console.error('Failed to fetch trades');
    }
  } catch (error) {
    console.error('Error fetching trades:', error.response ? error.response.data : error.message);
  }
}

// Example usage
const market_code = 'BTC-USD-SWAP-LIN';
const limit = 100;
const start_time = Date.now() - 24 * 60 * 60 * 1000; // 24 hours ago
const end_time = Date.now();
fetchTrades(market_code, limit, start_time, end_time);

File: Rest-Api-V3/trades/trades.py
import os
import requests
import hmac
import hashlib
import base64
import time
from dotenv import load_dotenv

load_dotenv()

def fetch_trades(market_code, limit, start_time, end_time):
    api_key = os.getenv('API_KEY')
    secret_key = os.getenv('API_SECRET').encode('utf-8')
    ts = time.strftime('%Y-%m-%dT%H:%M:%S', time.gmtime())
    nonce = str(int(time.time() * 1000))
    method = "/v3/trades"
    api_url = "api.ox.fun"

    query_string = f"marketCode={market_code}&limit={limit}&startTime={start_time}&endTime={end_time}"
    msg_string = f"{ts}\n{nonce}\nGET\n{api_url}\n{method}\n{query_string}"

    sign = base64.b64encode(hmac.new(secret_key, msg_string.encode('utf-8'), hashlib.sha256).digest()).decode('utf-8')

    headers = {
        'Content-Type': 'application/json',
        'AccessKey': api_key,
        'Timestamp': ts,
        'Signature': sign,
        'Nonce': nonce
    }

    url = f"https://{api_url}{method}?{query_string}"

    try:
        response = requests.get(url, headers=headers)
        response.raise_for_status()
        data = response.json()
        if data.get('success'):
            print('Trades Data:', data['data'])
        else:
            print('Failed to fetch trades')
    except requests.exceptions.RequestException as error:
        print('Error fetching trades:', error)

# Example usage
market_code = 'BTC-USD-SWAP-LIN'
limit = 100
start_time = int(time.time() * 1000) - 24 * 60 * 60 * 1000  # 24 hours ago
end_time = int(time.time() * 1000)
fetch_trades(market_code, limit, start_time, end_time)